### ISSUE TITLE
(🐞) mypy thinks `reveal_type` is defined

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#12285]
```py
reveal_type
not_defined
```
```
main.py:2:1: error: Name "not_defined" is not defined  [name-defined]
Found 1 error in 1 file (checked 1 source file)
```

```py
reveal_type
```
actual:
```
Success: no issues found in 1 source file
```
expected:
```
main.py:1:1: error: reveal_type must be called [misc]
```

**Comment by hauntsaninja** [comment]
This is intentional and documented. mypy treats reveal_type like a builtin

**Comment by KotlinIsland** [comment]
@hauntsaninja Thanks for the quick response!, but I think you might have mis-understood my bug report. I'm well aware of how `reveal_type` works, this issue I'm reporting is that when `reveal_type` is called (`reveal_type(1)`) mypy reports a note and exits with status 1, but when `reveal_type` is used as a value without being called mypy doesn't report any error or note.

I've updated the OP to clarify.

**Comment by hauntsaninja** [comment]
Okay, I can see how a non-zero exit code would be nice to have there, but this is low priority for me.

That said, maybe we shouldn't have non-zero exit status code for use of `typing.reveal_type` in 3.11 cc @JelleZijlstra

**Comment by JelleZijlstra** [comment]
I think the exit status issue is muddling the waters here. That's an unrelated change.

The original request was for an error if you use `reveal_type` without calling it. I guess we could do that but it doesn't seem worth the effort.

**Comment by KotlinIsland** [comment]
I think that exit 1 on `typing.reveal_type` is still good, it's completely useless and just gonna spam up the output of runs so mypy failing would be good to stop committing of code like that would be good IMO.
